pippy/PipelineDriver.py

```diff
@@ -1433,6 +1435,17 @@ def forward(self, *args, **kwargs):
         if DEBUG:
             self._check_stages_cleanup()
 
+        if self.pipe.has_loss_and_backwards:
+            local_results = []
+            last_grads = []
+            for local_result in local_results_and_last_grads:
+                local_results.append(local_result[0])
+                last_grads.append(local_result[1])
+
+            self.last_grads = last_grads
+        else:
+            local_results = local_results_and_last_grads
+
```

Comment:
nit: (ignore if you need to land first)
When we have a chance, shall we add a comment here explaining the purpose of the code block below? Thanks!